doc/userguide/upgrade.rst

```diff
@@ -30,6 +30,43 @@ by the ones Suricata supplies.
 Major updates include new features, new default settings and often also
 remove features.
 
+Suricata Configuration Defaults
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Suricata has a non-breaking/smooth upgrading policy when introducing new default
+settings for major version upgrades.
+
+This means that when a new default is introduced in a new major version, Suricata
+will issue warnings when the configuration setting is not explicitly present in
+the suricata.yaml file, to ensure that users will know that when the next major
+update happens, the default behavior will be applied regardless of having that
+setting defined in the configurations.
```

Comment:
Any examples of this currently happening?  Running with a very minimal configuration file is very much desired, and ideally it shouldn't lead to a load of warnings.

Reply:
When I run master-6.0.x without changing the yaml:
```
[92799] 8/12/2022 -- 17:58:00 - (app-layer-detect-proto.c:1970) <Warning> (AppLayerProtoDetectConfProtoDetectionEnabled) -- [ERRCODE: SC_ERR_CONF_YAML_ERROR(242)] - App-Layer protocol sip enable status not set, so enabling by default. This behavior will change in Suricata 7, so please update your config. See ticket #4744 for more details.
[92799] 8/12/2022 -- 17:58:00 - (app-layer-detect-proto.c:1970) <Warning> (AppLayerProtoDetectConfProtoDetectionEnabled) -- [ERRCODE: SC_ERR_CONF_YAML_ERROR(242)] - App-Layer protocol mqtt enable status not set, so enabling by default. This behavior will change in Suricata 7, so please update your config. See ticket #4744 for more details.
[92799] 8/12/2022 -- 17:58:00 - (app-layer-detect-proto.c:1970) <Warning> (AppLayerProtoDetectConfProtoDetectionEnabled) -- [ERRCODE: SC_ERR_CONF_YAML_ERROR(242)] - App-Layer protocol rdp enable status not set, so enabling by default. This behavior will change in Suricata 7, so please update your config. See ticket #4744 for more details.
```